pkg/asset/machines/worker.go

```diff
@@ -179,13 +179,28 @@ func defaultVSphereMachinePoolPlatform() vspheretypes.MachinePool {
 	}
 }
 
-func defaultPowerVSMachinePoolPlatform() powervstypes.MachinePool {
-	return powervstypes.MachinePool{
+func defaultPowerVSMachinePoolPlatform(ic *types.InstallConfig) powervstypes.MachinePool {
+	var (
+		defaultMp powervstypes.MachinePool
+		sysTypes  []string
+		err       error
+	)
+
+	defaultMp = powervstypes.MachinePool{
 		MemoryGiB:  32,
 		Processors: intstr.FromString("0.5"),
 		ProcType:   machinev1.PowerVSProcessorTypeShared,
 		SysType:    "s922",
 	}
+
+	sysTypes, err = powervstypes.AvailableSysTypes(ic.PowerVS.Region)
+	if err != nil {
+		defaultMp.SysType = sysTypes[0]
+	} else {
+		logrus.Warnf("For given region %v, AvailableSysTypes returns %v", ic.PowerVS.Region, err)
+	}
```

Comment:
Isn't the logic here reversed?
```suggestion
	if err == nil {
		defaultMp.SysType = sysTypes[0]
	} else {
		logrus.Warnf("For given region %v, AvailableSysTypes returns %v", ic.PowerVS.Region, err)
	}
```

Reply:
Weird!  I know I've tested this code in mad02 which should only have a non-922 system type.
Yes, the logic is reversed.